src/ucp/core/ucp_rkey.inl

```diff
@@ -45,7 +45,7 @@ ucp_rkey_get_tl_rkey(ucp_rkey_h rkey, ucp_md_index_t rkey_index)
         return UCT_INVALID_RKEY;
     }
 
-    ucs_assert(rkey_index < ucs_popcount(rkey->md_map));
+    ucs_assert(rkey_index < UCP_MAX_MDS);
```

Comment:
looks incorrect, rkey_index is supposed to be `rkey->tl_rkey` array index

Reply:
rkey->md_map 0x10 dst_md_index 7 rkey_index 1

Reply:
then something is weird - it should not fail since 1 <= ucs_popcount(0x10)=1

Reply:
seems the issues is that dst_md_index is not in rkey->md_map

Reply:
I checkout out an earlier version of ucx that still seemed to work: what I found was that the logic in ucp_rndv_zcopy_get_lane() lead to an rkey_index of 255, and because of that the if statement in ucp_rkey_get_tl_rkey 
```
if (rkey_index == UCP_NULL_RESOURCE) {
        return UCT_INVALID_RKEY;
    }
```
was kicking in.

Reply:
can you try returning UCP_NULL_RESOURCE from ucp_rndv_get_rkey_index if (!(UCS_BIT(md_index) & rkey->md_map)) ?

Reply:
md_index or dst_md_index? I assume its the second one.  Tests are running right now, but looks good so far.

Reply:
@yosefe This solution seems to work, thank you! I have update the pr accordingly.